src/main/java/io/cdap/plugin/google/shopping/content/source/ProductWritable.java

```diff
@@ -0,0 +1,42 @@
+package io.cdap.plugin.google.shopping.content.source;
+
+import com.google.gson.Gson;
+
+import org.apache.hadoop.io.Writable;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+/**
+ * The {@link Writable} interface for {@link ProductWrapper}.
+ */
+public class ProductWritable implements Writable {
```

Comment:
This class is no longer needed, since it will not get sent over the network to any other node in the cluster. This is because it immediately gets transformed in the batch source into a StructuredRecord.

You can change the InputFormat and RecordReader to output Product objects directly.